design/010-new-dataset-fields/010-new-dataset-fields.md

```diff
@@ -0,0 +1,130 @@
+- Start Date: 22-07-2022
+- Status: Proposed
+
+# New metadata fields for datasets
+
+## Summary
+
+Add standard metadata fields to make Renku datasets more FAIR.
+
+## Motivation
+
+As a Renku user, I want to easily annotate my datasets with standard metadata. Renku datasets support custom annotations as JSON-ld files. Although these annotations are available in the project graph, they are not in the global KG. Furthermore, the process of adding custom metadata is tedious and error prone (requires writing JSON-ld by hand). Currently, some crucial default metadata fields (e.g. license) are missing from datasets.
+
+Datasets metadata should support common fields required for FAIR-ification.
+
+## Design Detail
+
+### User research data
+
+We can leverage information from 3 sources to identify potential fields. In order of decreasing priority: FAIR recommendations, User feedback, User persona.
+
+**FAIR recommendations**
+
+* F1: (Meta)data are assigned a globally unique and persistent identifier
+  + _Ideally we would use something like ROR for **affiliation**_
+* R1.1: data are released with a clear and accessible data usage **license**
+
+**Feedback**
+
+* I would like to know the total dataset **size** to adjust session ressources and avoid running out of disk space
+* Dataset imports are slow
+  + _A **size** attribute may help by showing an ETA or even “large dataset” warning in the future._
+* I would like to search entities by **affiliation** and **topic** in the search page
+
+**Persona**
+
+* No visibility on my project
+  + _Unique **affiliation** (ROR, ORCID) + **topic** may help with that_
+
+### Proposed fields
+
+The following fields can be identified from the data above. They are listed in order of priority with the following logic: FAIR recommendation > user feedback > user persona.
+
+**License**
+
+Data usage license.
+
+* **Reason:** FAIR compliance
+* **Mapping:** [license](https://schema.org/license)
+* **Value:** [URL](https://schema.org/URL) from https://spdx.org/licenses
+* **Example:** `https://schema.org/license: https://spdx.org/licenses/MIT.html`
+
+**Affiliation**
+
+For visibility and citability, affiliations should have a unique identifier. Affiliation already exists as a property of the dataset's creator. Ideally we could add a unique identifier to it. Rather than a property of the Dataset, this may be a property of the creator's `affiliation` and would also be available for projects.
```

Comment:
It sounds like we'd want a new global node type that can be shared in many projects/datasets. That might be a bit larger effort.

The main downside I see of ROR (or at least its REST API) is that it would add internet connectivity as a requirement to some dataset commands that currently don't have that. And it might be down or have other issues like a schema update, which could then break functionality on our end.
Having it as an extra, optional property would be fine, alongside a unique ID generated by us. Though that somewhat defeats the purpose.

Reply:
I agree that affiliation should remain optional. Would it make sense to have "identifier" (ROR id) as an optional property of the organization to avoid making internet connectivity a hard dependency. Maybe with something like this:
1. User adds creator with an affiliation using `renku {project,dataset} {create,edit}`
2. renku-python tries to send query to ror.org
  + :x: (no internet connection or server down) -> identifier is not added to the organization and Renku warns the user, e.g.: :warning: Renku could not connect to the internet to associate a unique identifier with [affiliation].
  + :heavy_check_mark:  -> the ror ID is added as "identifier" to the organization

Reply:
The other option (or maybe in addition to the above?) would be to try and fill this in when generating triples in the KG using rp. We can assume to have an internet connection there (though ROR might still be down).

Reply:
We likely need a user action to disambiguate ROR results. Given an input string, the ROR API may return multiple affiliations and it would be risky to assume that the first result is correct. On the command line, the user would at least be able to select among a few options (like for `renku rollback`).

Is there currently not a global user/person node type ? The best would probably be to interactively set the affiliation once (maybe in the UI) and have the ROR id (and ORCID) associated with them in the KG. Then instead of querying ROR for every project or dataset, the user metadata could just be fetched from there?

Reply:
Affiliation should be a property of the user in Renku. It shouldn't need to be added on every action or every time triples are generated. I think it's a bit out of scope for the proposal here, but definitely something we need to do. I like the ROR idea (minus the caveats mentioned by @Panaetius) - lets keep it for a proposal on how to properly manage user metadata.